### PR TITLE
Improve silent authentication doc

### PR DESCRIPTION
**Pull request by rolodato**
Describes cases for regular webapps as well as SPAs

### CODE REVIEW

File: articles/api-auth/tutorials/silent-authentication.md [hunk 1]
```diff
@@ -1,45 +1,90 @@
...
 
-When you authenticate a user to your application, you would normally redirect them to the [`/authorize` endpoint of Auth0's authentication API](/api/authentication#social).
+## Refreshing expired tokens
+
+Access tokens are opaque to clients. This means that clients are unable to inspect the contents of access tokens to determine their expiration date.
+
+There are two options to determine when an access token expires:
+
+1. Read the `expires_in` response parameter returned by Auth0
+2. Ignore expiration dates altogether. Instead, try to renew the access token if rejected by the API.
```
Comment (twistedstream) [inline articles/api-auth/tutorials/silent-authentication.md]: I would rephrase the end of this sentence from:

> if rejected by the API.

to: 

> if your API rejects a request from the client (eg. with a `401`).

Just so its clear which API we're talking about (i.e. not Auth0's API).

File: articles/api-auth/tutorials/silent-authentication.md [hunk 2]
```diff
@@ -1,45 +1,90 @@
...
+When using the [Authorization Code Grant](/api-auth/grant/code), it is returned to the backend server when performing the authorization code exchange.
 
-* If the user is still authenticated, Auth0 will call back to your application immediately with renewed ID and access tokens.
-* If the user is no longer authenticated or had never authenticated, Auth0 will call back to your application with an error.
+The `expires_in` parameter indicates how many seconds the access token will be valid for, and can be used to anticipate expiration of the access token.
+When the access token has expired, silent authentication can be used to retrieve a new one without user interaction, assuming the user's SSO session has not expired.
 
-You can use the [`renewAuth` method from auth0.js](https://github.com/auth0/auth0.js#api) to authenticate users with `prompt=none` in an iframe.
-This lets you keep users logged in without needing to refresh or redirect away from your application.
+In the case of single-page applications, the [`renewAuth` method from auth0.js](https://github.com/auth0/auth0.js#api) can be used to perform silent authentication within an iframe.
```
Comment (twistedstream) [inline articles/api-auth/tutorials/silent-authentication.md]: I would change the end of this sentence:

> within an iframe.

to something like:

> within a hidden iframe, which results in no UX disruption at all.

**Comment by rolodato** [comment]
Updated with feedback from @twistedstream.

File: articles/api-auth/tutorials/silent-authentication.md [hunk 3]
```diff
@@ -1,45 +1,87 @@
...
-The `expires_in` parameter is returned by Auth0 as a hash parameter following a successful authentication.
-For example, Auth0 could call your callback URL with the following parameters:
+If the user was already logged in via SSO, Auth0 will respond exactly as if the user had authenticated manually through the SSO login page.
 
-```text
-https://myapp.example.com/
-    #access_token=...&
-    id_token=...&
-    token_type=Bearer&
+For example, when using the [Authorization Code Grant](/api-auth/grant/authorization-code) (`response_type=code`, used for regular web applications), Auth0 will respond with an authorization code that can be exchanged for an ID token and optionally an access token:
```
Comment (mpaktiti) [inline articles/api-auth/tutorials/silent-authentication.md]: @rolodato Isn't this the other way around? -> `Auth0 will respond with an authorization code that can be exchanged for an ID token and optionally an access token`
